multicollector.go

```diff
@@ -1,16 +1,51 @@
 package main
 
 import (
+	"cmp"
 	"context"
+	"fmt"
 	"log"
+	"maps"
 	"runtime"
+	"slices"
+	"strings"
 	"sync"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"golang.org/x/sync/errgroup"
 )
 
+func formatWarnings(warnings map[warningCategory][]error) string {
+	categories := slices.SortedFunc(maps.Keys(warnings), func(a, b warningCategory) int {
+		return cmp.Compare(a.String(), b.String())
+	})
```

Comment:
Why is sorting needed?

Reply:
To have stable output. Iteration order over maps is randomized. If there are warnings from multiple categories they'd be logged in unpredictable order (within a single log message).